src/main.zig

```diff
@@ -938,6 +942,13 @@ fn buildOutputType(
                         install_name = args_iter.next() orelse {
                             fatal("expected parameter after {s}", .{arg});
                         };
+                    } else if (mem.startsWith(u8, arg, "--compress-debug-sections=")) {
+                        const param = arg["--compress-debug-sections=".len..];
+                        linker_compress_debug_sections = std.meta.stringToEnum(link.CompressDebugSections, param) orelse {
+                            fatal("expected --compress-debug-sections=[none|zlib], found '{s}'", .{param});
+                        };
+                    } else if (mem.eql(u8, arg, "--compress-debug-sections")) {
+                        linker_compress_debug_sections = link.CompressDebugSections.zlib;
```

Comment:
@andrewrk I don't think we should be doing this. If `--compress-debug-sections` is specified alone, we should not be implying `zlib`, because in the future we may have a better compression algorithm, but we will be stuck with the default `zlib`, which seems like a good idea today, but not in the far future.

Also, neither clang, nor gcc accept such parameter: they require the compression algorithm:

```
motiejus@motiejus-20NYS8QG0B:/code/bazel-zig-cc/test/c$ clang-14 -Wl,--compress-debug-sections -g main.c -o main
/usr/bin/ld: invalid --compress-debug-sections option: `/tmp/main-de1faa.o'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
motiejus@motiejus-20NYS8QG0B:/code/bazel-zig-cc/test/c$ clang-14 -Wl,--compress-debug-sections=zlib -g main.c -o main
motiejus@motiejus-20NYS8QG0B:/code/bazel-zig-cc/test/c$ 
**motiejus@motiejus-20NYS8QG0B:/code/bazel-zig-cc/test/c$ gcc -Wl,--compress-debug-sections=zlib -g main.c -o main
motiejus@motiejus-20NYS8QG0B:/code/bazel-zig-cc/test/c$ gcc -Wl,--compress-debug-sections -g main.c -o main
/usr/bin/ld: invalid --compress-debug-sections option: `/tmp/cccg5gZi.o'
collect2: error: ld returned 1 exit status
motiejus@motiejus-20NYS8QG0B:/code/bazel-zig-cc/test/c$ gcc -Wl,--compress-debug-sections=zlib -g main.c -o main
motiejus@motiejus-20NYS8QG0B:/code/bazel-zig-cc/test/c$ gcc -Wl,--compress-debug-sections -g main.c -o main
/usr/bin/ld: invalid --compress-debug-sections option: `/tmp/cccg5gZi.o'
collect2: error: ld returned 1 exit status
motiejus@motiejus-20NYS8QG0B:/code/bazel-zig-cc/test/c$ 
```

Reply:
For those curious, LLVM is adding zstd: https://github.com/llvm/llvm-project/commit/449f2ca146dde0ef14c9837f71d4a0be92ca4843